modules/fundamental/src/vulnerability/service/test.rs

```diff
@@ -423,3 +424,58 @@ async fn vulnerability_queries(ctx: &TrustifyContext) -> Result<(), anyhow::Erro
 
     Ok(())
 }
+
+#[test_context(TrustifyContext)]
+#[test(actix_web::test)]
+async fn analyze_purls(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
+    let service = VulnerabilityService::new();
+
+    ctx.ingest_documents(["osv/RUSTSEC-2021-0079.json", "osv/GHSA-2ccf-ffrj-m4qw.json"])
+        .await?;
+
+    let result = service.analyze_purls(vec![], &ctx.db).await?;
+
+    assert!(result.is_empty());
+
+    let fail = ["this is not valid", "pkg:npm/missing.version"];
+
+    let result = service.analyze_purls(fail.to_vec(), &ctx.db).await;
+
+    assert!(result.is_err());
+
+    match result {
+        Err(Error::BadRequest(msg)) => assert!(msg.contains("Invalid purls received")),
+        _ => panic!("Expected BadRequest error"),
+    }
```

Comment:
```suggestion
    assert!(result.is_err());
```
I don't love verifying error strings in tests. Feels brittle to me. For what we're doing, I think `.is_err()` is enough.